geodetic_utils/src/set_gps_reference_node.cpp

```diff
@@ -30,27 +43,37 @@ void gps_callback(const sensor_msgs::NavSatFixConstPtr & msg)
   g_lat_ref += msg->latitude;
   g_lon_ref += msg->longitude;
   g_alt_ref += msg->altitude;
+  g_yaw_ref += g_latest_imu_msg.orientation.z;
+
+  ROS_INFO("Current measurement: %f, %f, %f, %f", msg->latitude, msg->longitude, msg->altitude, g_latest_imu_msg.orientation.z);
 
-  ROS_INFO("Current measurement: %f, %f, %f", msg->latitude, msg->longitude, msg->altitude);
+  if (!g_got_imu) {
+  ROS_WARN_STREAM_THROTTLE(1, "No IMU data yet");
+  return;
+}
 
   if (count == g_its) {
     if (g_mode == MODE_AVERAGE) {
       g_lat_ref /= g_its;
       g_lon_ref /= g_its;
       g_alt_ref /= g_its;
-    } else {
+      g_yaw_ref /= g_its;
+    }
+    else {
       g_lat_ref = msg->latitude;
       g_lon_ref = msg->longitude;
       g_alt_ref = msg->altitude;
+      g_yaw_ref = g_latest_imu_msg.orientation.z;
```

Comment:
Why store only the z component of the rotation quaternion? @raghavkhanna @marija-p